File: app/src/main/java/com/shuoye/video/fragment/home/TimeLineFragment.java
package com.shuoye.video.fragment.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.alibaba.fastjson.JSONObject;
import com.shuoye.video.databinding.FragmentTimeLineBinding;
import com.shuoye.video.pojo.TimeLine;

import org.jetbrains.annotations.NotNull;

import java.util.List;


/**
 * 新番时间表
 *
 * @author shuoye
 */
public class TimeLineFragment extends Fragment {
    private final List<TimeLine> data;
    FragmentTimeLineBinding binding;
    TimeLineAdapter adapter;

    public TimeLineFragment(List<TimeLine> data) {
        this.data = data;
    }


    @Override
    public View onCreateView(@NotNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentTimeLineBinding.inflate(inflater);
        adapter = new TimeLineAdapter(data);
        binding.recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        binding.recyclerView.setAdapter(adapter);
        return binding.getRoot();
    }
}

File: app/src/main/java/com/shuoye/video/ui/home/TimeLineFragment.kt
package com.shuoye.video.ui.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.shuoye.video.databinding.FragmentRecyclerViewBinding
import com.shuoye.video.ui.home.adapters.TimeLineAdapter
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

/**
 * TODO
 * @program Video
 * @ClassName TimeLineFragment
 * @author shuoye
 * @create 2021-10-24 15:20
 **/
@AndroidEntryPoint
class TimeLineFragment(val wd: Int) : Fragment() {
    private val viewModel: TimeLineViewModel by viewModels()
    private val adapter = TimeLineAdapter()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val binding = FragmentRecyclerViewBinding.inflate(inflater, container, false)
        context ?: return binding.root

        binding.recyclerView.layoutManager = LinearLayoutManager(context)
        binding.recyclerView.adapter = adapter
        lifecycleScope.launch {
            viewModel.getTimeLines(wd).collectLatest {
                adapter.submitData(it)
            }
        }
        return binding.root
    }

    fun initView() {
    }
}